src/main/java/org/wildfly/security/auth/client/ElytronXmlParser.java

```diff
@@ -1394,6 +1419,291 @@ static Module parseModuleRefType(ConfigurationXMLStreamReader reader) throws Con
         throw reader.unexpectedDocumentEnd();
     }
 
+    /**
+     * Parse an XML element of type {@code set-bearer-token} from an XML reader.
+     *
+     * @param reader the XML stream reader
+     * @param bearerTokensMap the map of bearer tokens to use
+     * @return the bearer token entry factory
+     * @throws ConfigXMLParseException if the resource failed to be parsed
+     */
+    public static SecurityFactory<CallbackHandler> parseBearerTokenRefType(ConfigurationXMLStreamReader reader, final Map<String, SecurityFactory<CallbackHandler>> bearerTokensMap) throws ConfigXMLParseException {
```

Comment:
I think the visibility looks a bit too high here. We don't have a need for this to be called from elsewhere do we?